File: java/zanzibar.java
import java.util.ArrayList;
import java.util.Scanner;

public class zanzibar {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int T = scan.nextInt();
		while (T-- > 0) {
			ArrayList<Integer> turtles = new ArrayList<>();
			int t;
			while ((t = scan.nextInt()) != 0)
				turtles.add(t);
			int import_turtles = 0;
			for (int j = 1; j < turtles.size(); j++)
				import_turtles += Math.max(turtles.get(j) - turtles.get(j - 1) * 2, 0);
			System.out.println(import_turtles);
		}
		scan.close();
	}
}

File: ts/src/zanzibar.ts
import readline = require('readline')

function getImportedTurtles (turtles: number[]): number {
  let imported = 0
  for (let i = 1; i < turtles.length; i++) {
    imported += Math.max((turtles[i] ?? 0) - (turtles[i - 1] ?? 0) * 2, 0)
  }
  return imported
}

let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      const turtles = line.split(' ').map(Number)
      turtles.pop()
      console.log(getImportedTurtles(turtles))
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
